### PR TITLE
feat(components): localized date formatter

### PR DESCRIPTION
**Pull request by darryltec**
## Motivations

Now that we are starting to support localization, date and time formatters needs to be localized too

### Added

- `formatTime` and `formatDate` in the `useAtlantisI18n` hook to automatically localized the date or time string depending on your settings in Atlantis Context

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

- Implement Atlantis context on storybook preview and change locale to "es"
- Check InputDate/Mobile/Basic, it should have a different date format
- Check InputTime/Mobile/Basic, it should have a different time format

```diff
diff --git a/.storybook/preview.tsx b/.storybook/preview.tsx
index 7df4c8d5..ae1f6f40 100644
--- a/.storybook/preview.tsx
+++ b/.storybook/preview.tsx
@@ -1,6 +1,7 @@
 import React from "react";
 import { Text } from "@jobber/components/Text";
 import { Content } from "@jobber/components/Content";
+import { AtlantisContext, defaultValues } from "@jobber/components-native";
 import { tokens } from "@jobber/design/foundation";
 import { Header } from "./components/Header";
 import { InlineCode } from "./components/InlineCode";
@@ -71,4 +72,5 @@ export const decorators = [
     </SBProvider>
   ),
   (Story, context) => <StoryDetails Story={Story} context={context} />,
+  (Story) => <AtlantisContext.Provider value={{...defaultValues, locale: "es"}}>{Story()}</AtlantisContext.Provider>,
 ];
```

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by ZakaryH** [comment]
I'm a bit nervous about the test coverage here

when I run the tests in `useAtlantisI18n.test.ts` along with a `console.log(timeZone)` in the `dateFormatter` method, it's running as `UTC`, meaning that `utcToZonedTime` isn't doing a whole lot with this set of data because it'll be converting a utc time to.... well, utc

I admit that this is borderline testing the functionality of `utcToZonedTime`, and that one reason it could be failing is that there's something wrong with that library's method, or the `date-fns` `format` method which is the other piece we're using here - but I feel like the combination of those two that we are using to achieve the functionality is worth testing

could we please add some tests other timezones and verify the output is correct? 

so one timezone with a negative offset which can be anything in North America (that observes DST, maybe don't choose Phoenix since that could get weird.... or maybe do? idk)

then another timezone with a positive offset, so maybe something in Asia or Australia.

thinking about how to do this, you'd probably want to use `timezone-mock` because by default in Jest, the timezone returned whenever you ask the machine for its tz, it'll be UTC like what we're probably seeing in these tests

https://github.com/Jimbly/timezone-mock

hopefully that's chill to add a dev dependency here, but if not just lmk!

**Review by ZakaryH — CHANGES_REQUESTED** [review]
would love to have some more test coverage when timezones are not UTC

left a comment with some thoughts, happy to chat about it!

**Comment by darryltec** [comment]
@ZakaryH oh good! was thinking of testing another time zone after our chat on Monday so yeah, will look into that bad boi

File: packages/components-native/src/hooks/useAtlantisI18n/useAtlantisI18n.ts [hunk 1]
```diff
@@ -1,21 +1,57 @@
...
+  /**
+   * The set locale based on the AtlantisContext.
+   */
   readonly locale: string;
+
+  /**
+   * Returns the translated string depending on the locale. This accepts a 2nd
+   * param for string interpolation.
+   */
   readonly t: (
```
[the remark this answers is not in the document]
Comment (darryltec) [reply packages/components-native/src/hooks/useAtlantisI18n/useAtlantisI18n.ts:17]: That's a common pattern.

https://www.i18next.com/overview/api#t
https://guides.rubyonrails.org/i18n.html#the-public-i18n-api

[the remark this answers is not in the document]
Comment (MichaelParadis) [reply packages/components-native/src/hooks/useAtlantisI18n/useAtlantisI18n.ts:17]: @darryltec thanks for the context. I don't like it but that is a me thing.

**Review by ZakaryH — APPROVED** [review]
awesome, thanks for adding those tests!

LGTM

**Pull request merged**